src/main/java/dev/nmarulo/depensaapp/security/JwtAuthenticationEntryPoint.java

```diff
@@ -0,0 +1,50 @@
+package dev.nmarulo.depensaapp.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.nmarulo.depensaapp.commons.dtos.ErrorRes;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.jwt.JwtException;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationEntryPoint;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationEntryPoint implements AuthenticationEntryPoint {
+    
+    private final AuthenticationEntryPoint delegate = new BearerTokenAuthenticationEntryPoint();
+    
+    private final ObjectMapper mapper;
+    
+    @Override
+    public void commence(HttpServletRequest request,
+                         HttpServletResponse response,
+                         AuthenticationException authException) throws IOException, ServletException {
+        
+        this.delegate.commence(request, response, authException);
+        
+        if (authException.getCause() instanceof JwtException jwtException) {
+            var problemDetail = ProblemDetail.forStatus(HttpStatus.UNAUTHORIZED);
+            
+            problemDetail.setType(URI.create("https://tools.ietf.org/html/rfc6750#section-3.1"));
+            problemDetail.setInstance(URI.create(request.getRequestURI()));
+            problemDetail.setProperty("errorCode", "invalid_token");
+            problemDetail.setProperty("description", jwtException.getMessage());
+            problemDetail.setTitle("Invalid Token");

```

Comment:
As this information is already in the header, it is enough for me to set status, detail (exception message) and title fields.

Reply:
Makes sense I guess, however I'd still prefer that the `problemDetail.setInstance(URI.create(request.getRequestURI()));` still be there? That way even if it is in the header the response body would have clear indication as to in which URL the error occurred?

Reply:
I honestly don't know what use I could make of having the “RequestURI” in the body of the response, right now the web part at best shows the “detail” field to the user.

For the moment just filling in the 3 fields I mentioned is enough for me, if you want to leave extra information, ok, no problem.

Reply:
Sure, done.

Reply:
I forgot to mention that there is a field called `detail` in the `ProblemDetail` class, so it is not necessary to create the `description` property. There is even a static method `.forStatusAndDetail(...)`.

Reply:
Good catch! Done!